lib/options.go

```diff
@@ -44,6 +45,86 @@ const DefaultScenarioName = "default"
 // nolint: gochecknoglobals
 var DefaultSummaryTrendStats = []string{"avg", "min", "med", "max", "p(90)", "p(95)"}
 
+type DNSConfig struct {
+	TTL      null.String     `json:"ttl"`
+	Strategy NullDNSStrategy `json:"strategy"`
+	// FIXME: Valid is unused and is only added to satisfy some logic in lib.Options.ForEachSpecified(),
+	// otherwise it would panic with `reflect: call of reflect.Value.Bool on zero Value`.
+	Valid bool
+}
+
+//go:generate enumer -type=DNSStrategy -transform=kebab -trimprefix DNS -output dns_strategy_gen.go
+type DNSStrategy uint8
+
+type NullDNSStrategy struct {
+	DNSStrategy
+	Valid bool
+}
+
+const (
+	DNSFirst DNSStrategy = iota + 1
+	DNSRoundRobin
+	DNSRandom
+)
+
+const DefaultDNSConfigText = "ttl=inf,strategy=first"
+
+func DefaultDNSConfig() DNSConfig {
+	return DNSConfig{
+		TTL:      null.NewString("inf", false),
+		Strategy: NullDNSStrategy{DNSFirst, false},
+	}
+}
+
+func (c DNSConfig) MarshalJSON() ([]byte, error) {
+	strat := ""
+	if c.Strategy.IsADNSStrategy() {
+		strat = c.Strategy.String()
+	}
+	return json.Marshal(struct {
+		TTL      string `json:"ttl"`
+		Strategy string `json:"strategy"`
+	}{
+		TTL:      c.TTL.String,
+		Strategy: strat,
+	})
+}
+
+func (c *DNSConfig) UnmarshalJSON(data []byte) error {
+	params := make(map[string]interface{})
+	if err := json.Unmarshal(data, &params); err != nil {
+		return err
+	}
+	return c.unmarshalDNSConfig(params)
+}
+
+func (c *DNSConfig) UnmarshalText(text []byte) error {
+	if string(text) == DefaultDNSConfigText {
+		*c = DefaultDNSConfig()
+		return nil
+	}
+	params, err := strvals.Parse(string(text))
+	if err != nil {
+		return err
+	}
+	return c.unmarshalDNSConfig(params)
+}
+
+func (c *DNSConfig) unmarshalDNSConfig(params map[string]interface{}) error {
```

Comment:
```suggestion
func (c *DNSConfig) unmarshal(params map[string]interface{}) error {
```

Reply:
Yeah, but this isn't really making it worse. We'd likely need 90% of it even if we had a sane config system, as this is just marshalling code for the internal representation, which should handle both JSON and the strvals format.

Anyway, shortened the method name in f06ad67d.